### ISSUE TITLE
persistence: failed to append to trace: File exists (os error 17) during startup

### ISSUE THREAD

**Issue by philip-stoev** [MaterializeInc/materialize#7923]
### What version of Materialize are you using?

```
ed424518a
```

### What was the issue?

After Ctrl+Cing and restarting Materialize, I got the following output:

```
2021-08-19T10:37:06.646031Z  INFO coord::catalog: create table materialize.public.nation (u71)    
2021-08-19T10:37:06.652670Z  WARN persist::indexed::runtime: error running step: String("failed to append to trace: File exists (os error 17)")    
2021-08-19T10:37:06.663218Z  INFO coord::catalog: create index materialize.public.nation_primary_idx (u72)    
2021-08-19T10:37:06.673169Z  WARN persist::indexed::runtime: error running step: String("failed to append to trace: File exists (os error 17)")    
2021-08-19T10:37:06.683188Z  INFO coord::catalog: create index materialize.public.pk_nation_nationkey (u73)    
2021-08-19T10:37:06.695416Z  WARN persist::indexed::runtime: error running step: String("failed to append to trace: File exists (os error 17)")    
2021-08-19T10:37:06.705378Z  INFO coord::catalog: create index materialize.public.fk_nation_regionkey (u74)    
2021-08-19T10:37:06.720774Z  WARN persist::indexed::runtime: error running step: String("failed to append to trace: File exists (os error 17)")    
2021-08-19T10:37:06.731393Z  INFO coord::catalog: create table materialize.public.region (u75)    
2021-08-19T10:37:06.749485Z  WARN persist::indexed::runtime: error running step: String("failed to append to trace: File exists (os error 17)")    
2021-08-19T10:37:06.759890Z  INFO coord::catalog: create index materialize.public.region_primary_idx (u76)    
2021-08-19T10:37:06.781814Z  WARN persist::indexed::runtime: error running step: String("failed to append to trace: File exists (os error 17)")    
2021-08-19T10:37:06.792466Z  INFO coord::catalog: create index materialize.public.pk_region_regionkey (u77)    
2021-08-19T10:37:06.816630Z  WARN persist::indexed::runtime: error running step: String("failed to append to trace: File exists (os error 17)")    
2021-08-19T10:37:06.827465Z  INFO coord::catalog: create table materialize.public.part (u78)    
```
### Is the issue reproducible? If so, please provide reproduction instructions.

I only saw it once after loading the data from a DBT-3 dataset scale s0.001

**Comment by ruchirK** [comment]
this seems like a "working as expected". the write to trace is not guaranteed to be atomic, and a partial write to a trace batch is fine - we won't attempt to read the trace file on startup because its not in meta, then we might collide once on the file's id, but then we'll pick a new id, and things will proceed. The only thing missing from this story imo is that the partially written file never gets gc'd, and the error message could be a lot better

**Comment by danhhz** [comment]
This was fixed by #8667
